packages/react/src/components/Storage/FileUploader/Tracker/index.tsx

```diff
@@ -1,18 +1,115 @@
 import React from 'react';
-import { Card, View } from '../../../../primitives';
+import { humanFileSize } from '@aws-amplify/ui';
+import { TrackerProps } from '../types';
+import {
+  Card,
+  Flex,
+  View,
+  Image,
+  Text,
+  Button,
+  TextField,
+} from '../../../../primitives';
+import { CloseIcon, EditIcon, fileIcon } from '../Previewer/PreviewerIcons';
 
 export function Tracker({
-  children,
-}: {
-  children?: React.ReactNode;
-}): JSX.Element {
+  file,
+  hasImage,
+  url,
+  onChange,
+  onCancel,
+  name,
+}: TrackerProps): JSX.Element {
+  const [isEditing, setIsEditing] = React.useState(false);
+  if (!file) return null;
+
+  const { size } = file;
+
+  let icon = (
+    <View className="amplify-fileuploder--img-placeholder">{fileIcon}</View>
+  );
+
+  if (hasImage) {
+    icon = <Image alt="" maxHeight="100%" height="100%" src={url} />;
+  }
```

Comment:
Instead of using `let` and reassignment, can we set this through single `const` assignment (with ternary)?

Reply:
Done